src/main/scala/com/yotpo/metorikku/Metorikku.scala

```diff
@@ -1,14 +1,31 @@
 package com.yotpo.metorikku
 
-import com.yotpo.metorikku.configuration.job.ConfigurationParser
+import java.util.concurrent.{ScheduledThreadPoolExecutor, TimeUnit}
+
+import com.yotpo.metorikku.configuration.job.{ConfigurationParser, Periodic}
 import com.yotpo.metorikku.metric.MetricSet
 import org.apache.log4j.LogManager
 
 object Metorikku extends App {
   val log = LogManager.getLogger(this.getClass)
   log.info("Starting Metorikku - Parsing configuration")
   val session = Job(ConfigurationParser.parse(args))
-  runMetrics(session)
+
+  session.config.periodic match {
+    case Some(periodic) => {
+      executePeriodicTask(periodic)
+    }
+    case _ => runMetrics(session)
+  }
+
+  private def executePeriodicTask(periodic: Periodic) = {
+    val task = new Runnable {
+      def run() = runMetrics(session)
+    }
+    val ex = new ScheduledThreadPoolExecutor(1)
+    val initialDelay = 0
+    ex.scheduleAtFixedRate(task, initialDelay, periodic.getTriggerDurationInSeconds(), TimeUnit.SECONDS)
```

Comment:
What happens if it takes longer than the time of the trigger? it enters a queue? then we'll just have endless queue?

Reply:
It's not supposed to overflow:

https://stackoverflow.com/questions/43983993/why-is-the-queue-size-of-a-scheduledthreadpoolexecutor-always-0